src/basic/fileio.c

```diff
@@ -148,6 +148,28 @@ int write_string_stream_ts(
                         return -EBADF;
         }
 
+        if (flags & WRITE_STRING_VIRTUALFILE_SUPRESS_IF_POSSIBLE) {
+                _cleanup_free_ char *ret = NULL;
+
+                if (fd < 0) {
+                        fd = fileno(f);
+                        if (fd < 0)
+                                return -EBADF;
+                }
+
+                r = read_virtual_stream(fd, SIZE_MAX, &ret, NULL);
+                if (r > 0) {
+                        ret = delete_trailing_chars(ret, NEWLINE);
```

Comment:
you should probably ignore newlines both for the current and for the new value, i.e. delete the newlines on this argument, then check if the result is a prefix of the value you want to write, and then check if the remainder of that string only consists of newlines or is empty.